src/main/java/com/wow/server/controller/UserController.java

```diff
@@ -1,27 +1,61 @@
 package com.wow.server.controller;
 
-import java.util.List;
-
+import com.wow.server.data.model.User;
+import com.wow.server.data.repository.UserRepository;
+import com.wow.server.dto.UserDTO;
+import com.wow.server.dto.UserMinimalDTO;
+import com.wow.server.exception.DataNotFoundException;
+import com.wow.server.mapper.UserMapper;
+import com.wow.server.mapper.UserMinimalMapper;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.wow.server.data.model.User;
-import com.wow.server.data.repository.UserRepository;
+import java.util.List;
+import java.util.Optional;
 
+@Slf4j
 @RestController
+@RequestMapping("/api/users")
+@Tag(name = "Users API")
 public class UserController {
 
-	@Autowired
-	UserRepository userRepository;
-	
-	@RequestMapping (value= {"/api/users"}, method=RequestMethod.GET)
-	public ResponseEntity<List<User>> getUsers () {
-		
-		return ResponseEntity.ok(userRepository.findAll());
-		
-	}
-	
+    private final UserRepository userRepository;
+    private final UserMinimalMapper userMinimalMapper;
+    private final UserMapper userMapper;
+
+    @Autowired
+    public UserController(
+            UserRepository userRepository,
+            UserMinimalMapper userMinimalMapper,
+            UserMapper userMapper) {
+        this.userRepository = userRepository;
+        this.userMinimalMapper = userMinimalMapper;
+        this.userMapper = userMapper;
+    }
+
+    @GetMapping
+    @Operation(summary = "Returns list of all users with minimally required set of fields")
+    public List<UserMinimalDTO> getUsers() {
```

Comment:
Think this should rather by specific range, sorted by some specific critera
like top performers, top subscribers, etc
and paginated as well id more then 10?25?50?100? not sure what is the comfortable list length to scroll/navigate through on mobile and desktop browser

Reply:
This endpoint was not in original requirements. I'm trying to forecast needs here. Therefore input from @twinzom  and FE would be invaluable.

Reply:
> and paginated as well id more then 10?25?50?100?
That is something I specifically left out of scope. It is not hard to implement with current Spring tooling, but anyway I'd need to spent some time on this. In general I think for current amount of data, it is acceptable to not have any pagination, for future use it is "must have" thing. But again would need just some bit of effort.